src/index.ts

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+'use-strict';
+
+import {
+  Contents
+} from 'jupyter-js-services';
+
+import {
+  Widget
+} from 'phosphor-widget';
+
+import './index.css';
+
+/**
+ * A widget which hosts a file browser.
+ */
+export
+class FileBrowser extends Widget {
+
+  /**
+   * Create a new node for the file list.
+   */
+  static createNode(): HTMLElement {
+    var node = document.createElement('div');
+    node.innerHTML = (
+      '<div class="files_inner">' +
+        '<div class="files_header">Files</div>' +
+        '<div class="list_container"></div>' +
+      '</div>'
+    );
+    return node;
+  }
+
+  /**
+   * Construct a new file browser widget.
+   */
+  constructor(baseUrl: string, currentDir: string) {
+    super();
+    this.addClass('FileBrowser');
+    this._contents = new Contents(baseUrl);
+    document.addEventListener('mousedown', this, true);
+    this._currentDir = currentDir;
+  }
+
+  /**
+   * Get the onClick handler for the file browser.
+   */
+  get onClick(): (name: string, contents: any) => void {
+    return this._onClick;
+  }
+
+  /**
+   * Set the onClick handler for the file browser.
+   */
+  set onClick(cb: (name: string, contents: any) => void) {
+    this._onClick = cb;
+  }
+
+  /**
+   * Handle the events on the file browser.
```

Comment:
Suggest using the docstring here: https://github.com/phosphorjs/phosphor-splitpanel/blob/master/src/index.ts#L269.